ruby_event_store/lib/ruby_event_store/mappers/event_class_remapper.rb

```diff
@@ -0,0 +1,20 @@
+module RubyEventStore
+  module Mappers
+    class EventClassRemapper
+      def initialize(class_map)
+        @class_map = class_map
+      end
+
+      def dump(item)
+        item
+      end
+
+      def load(item)
+        item.merge(event_type: class_map.fetch(item.fetch(:event_type)) { item.fetch(:event_type) })
```

Comment:
Following might be simpler than requiring `fetch` on class map that we don't make use of really:
```
class_map[item.fetch(:event_type)] || item.fetch(:event_type)
```